projects/saturn/src/main/java/io/fairspace/saturn/services/search/AiSearchApp.java

```diff
@@ -0,0 +1,222 @@
+package io.fairspace.saturn.services.search;
+
+import lombok.extern.log4j.*;
+import nl.hyve.llm.LlmConversation;
+
+import org.json.JSONObject;
+
+import spark.Request;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import io.fairspace.saturn.services.BaseApp;
+
+import static io.fairspace.saturn.auth.RequestContext.getAccessToken;
+import static org.eclipse.jetty.http.MimeTypes.Type.APPLICATION_JSON;
+import static spark.Spark.get;
+import static spark.Spark.post;
+
+import java.io.FileWriter;
+
+@Log4j2
+public class AiSearchApp extends BaseApp {
+    final String CACHE_DIR = "./data/conversations/";
+
+    public AiSearchApp(String basePath) {
+        super(basePath);
+    }
+
+    @Override
+    protected void initApp() {
+        get("/newchat", "application/json", (req, res) -> {
```

Comment:
I would think of the API in  REST/HTTP API way:

- use HTTP DELETE method for deletion
- use HTTP POST method for any type of creation (with the GET '/newchat' we change the state, it is not even idempotent)
- not to use verbs in the path, use proper HTTP methods 


We could consider API like this:

- GET /chat - get all conversations
- POST /chat - current /newchat
- GET /chat/{id} - current /history/:id NOTE: I did not gat the difference between `'/history/:id'` and `'/conversation/:id'`, could you elaborate please?
- POST /chat/{id}/query
- DELETE /chat/{id}

There is one more in the code POST '/' - I don't get it, please elaborate too